low-level/actuators/impl/centos_7/systemd_service.py

```diff
@@ -51,117 +53,198 @@ def __init__(self):
-                self._manager.StartUnit(self._service_name, 'replace')
+            result = {}
+            is_err_response = False
+            if self._service_request in ['restart', 'start']:
+                # Before restart/start the service, check service state.
+                # If it is not active or activating then only process
+                # restart/start request.
+                service_state  = self._service.get_state(self._service_name)
+                state = service_state.state
+                if state not in ['active','activating']:
+                    if self._service_request == "restart":
+                        self._service.restart(self._service_name)
+                    elif self._service_request == "start":
+                        self._service.start(self._service_name)
+                    # Ensure we get an "active" state and not "activating"
+                    service_state  = self._service.get_state(self._service_name)
+                    state = service_state.state
+                    max_wait = 0
+                    while state != "active":
+                        logger.debug("%s status is activating, needs 'active' "
+                        "state after %s request has been processed, retrying"
+                        % (self._service_name, self._service_request))
+                        time.sleep(1)
+                        max_wait += 1
+                        if max_wait > 20:
+                            logger.debug("maximum wait - %s seconds, for "
+                                        "service restart reached." %max_wait)
+                            break
+                        service_state = self._service.get_state(self._service_name)
+                        state = service_state.state
+
+                else:
+                    is_err_response = True
+                    err_msg = ("Can not process %s request, for %s, as service "
+                    "is already in %s state." %(self._service_request,
+                                                    self._service_name, state))
+                    logger.error(err_msg)
+                    return (self._service_name, err_msg, is_err_response)
 
             elif self._service_request == "stop":
-                self._manager.StopUnit(self._service_name, 'replace')
+                self._service.stop(self._service_name)
 
             elif self._service_request == "status":
                 # Return the status below
-                state = self._get_activestate()
-                substate = self._get_active_substate()
-                self._log_debug("perform_request, state: %s, substate: %s" %
-                                (str(state), str(substate)))
-                return (self._service_name, state, substate)
+                service_status = self._service.get_state(self._service_name)
 
+            # TODO: Use cortx.utils Service class methods for
+            # enable/disable services.

```

Comment:
enable/disable from cortx-utils still giving error? If yes, let it be this way only

Reply:
yes, its still giving error.